app/models/trait.rb

```diff
@@ -6,9 +6,36 @@ class Trait < ActiveRecord::Base
   def self.create_new_trait_and_relation(new_factory, trait)
     new_trait = Trait.create(name: trait)
     REDIS.sadd("traits", { name: trait, factory_name: new_factory.name }.to_json)
-    new_trait_relation = new_factory.trait_relations.build
-    new_trait_relation.trait = new_trait
-    new_trait_relation.save
-    new_trait
+    TraitRelation.create_new_trait_relation(new_factory, new_trait)
   end
+
+  # TODO: write test
+  def self.create_new_traits_and_relations(factory, traits_names)
+    traits_names.each do |trait|
+      create_new_trait_and_relation(new_factory, trait)
+    end
+  end
+
+  # TODO: write test
+  # Create and register only not connected traits to factory.
+  def self.add_new_traits_and_relations(factory, trait_names)
+    trait_names.each do |trait|
+      if same_trait_exist?(trait, factory.name)
+        TraitRelation.create_new_trait_relation(factory, seach_existing_trait(trait, factory.name))
+      else
+        create_new_trait_and_relation(factory, trait)
+      end
+    end
+  end
+
+  private
```

Comment:
Lint/UselessAccessModifier: Useless `private` access modifier.<br>Style/EmptyLinesAroundAccessModifier: Keep a blank line before and after `private`.